portfolio/src/main/java/com/google/sps/servlets/UserLogin.java

```diff
@@ -0,0 +1,34 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/")
+public class UserLogin extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
```

Comment:
`urlToRedirectToAfterUserLogsOut` and `urlToRedirectToAfterUserLogsIn` are set to the same value, so consider using a single variable. 

Also consider making the variable an instance variable so you don't need to declare it every time the function is called.

Reply:
Done, thanks! I will have to change it back to two separate variables later since the urls will be different.